version: '3.8'

services:
    next_app:
        image: {{ env.IMAGE_URL }}
        restart: always
        ports:
            - '{{ env.UD_PORT }}:{{ env.UD_PORT }}'
        environment:
            - PORT={{ env.UD_PORT }}
            - DB_HOST=db
            - DB_PORT={{ env.UD_DB_PORT }}
            - DB_USER={{ env.UD_DB_USER }}
            - DB_PASSWORD={{ env.UD_DB_PASSWORD }}
            - DB_DATABASE={{ env.UD_DB_DATABASE }}
        depends_on:
            - db

    db:
        image: postgres:16-alpine
        environment:
            - POSTGRES_USER={{ env.UD_DB_USER }}
            - POSTGRES_PASSWORD={{ env.UD_DB_PASSWORD }}
            - POSTGRES_DB={{ env.UD_DB_DATABASE }}
            - PGPORT={{ env.UD_DB_PORT }}
        expose:
            - '{{ env.UD_DB_PORT }}'
        volumes:
            - ./data/db:/var/lib/postgresql/data
        healthcheck:
          test: ["CMD-SHELL", "pg_isready -U {{ env.UD_DB_USER }} -d {{ env.UD_DB_DATABASE }}"]
          interval: 10s
          timeout: 10s
          retries: 10
